src/ciphersuite.rs

```diff
@@ -7,19 +7,22 @@
 
 use crate::{
     errors::InternalPakeError,
-    group::Group,
     keypair::{Key, KeyPair},
+    password_to_curve::PasswordToCurve,
     slow_hash::SlowHash,
 };
-use generic_array::typenum::{U32, U64};
+
 use rand_core::{CryptoRng, RngCore};
 
 /// Configures the underlying primitives used in OPAQUE
-/// * Group: a finite cyclic group along with a point representation
-/// * KeyFormat: a keypair type composed of public and private components
-/// * SlowHash: a slow hashing function, typically used for password hashing
+/// * `Group`: a finite cyclic group along with a point representation, along
+///   with an extension trait PasswordToCurve that allows some customization on
+///   how to hash a password to a curve point. See `group::Group` and
+///   `password_to_curve::PasswordToCurve`.
+/// * `KeyFormat`: a keypair type composed of public and private components
+/// * `SlowHash`: a slow hashing function, typically used for password hashing
 pub trait CipherSuite {
-    type Group: Group<ScalarLen = U32, UniformBytesLen = U64>;
+    type Group: PasswordToCurve;
```

Comment:
Naming thing: Can we keep the top-level trait that is used in CipherSuite still called "Group", instead of PasswordToCurve? It does more than just turn a password into a curve point, it also does the group operations. Also, this should be kept more or less in sync with the lib.rs docs which will just refer to this as a group.